src/DynamicRupture/FrictionLaws/ThermalPressurization/ThermalPressurization.h

```diff
@@ -0,0 +1,153 @@
+#ifndef SEISSOL_THERMALPRESSURIZATION_H
+#define SEISSOL_THERMALPRESSURIZATION_H
+
+#include <array>
+
+#include "DynamicRupture/Misc.h"
+#include "DynamicRupture/Parameters.h"
+#include "Initializer/DynamicRupture.h"
+#include "Kernels/precision.hpp"
+
+namespace seissol::dr::friction_law {
+
+/**
+ * Logarithmic gridpoints as defined in Noda&Lapusta (14). These are the \f$\hat{l}\f$ for
+ * ThermalPressurization.
+ */
+template <size_t N>
+class GridPoints {
+  public:
+  GridPoints() {
+    for (size_t i = 0; i < N; ++i) {
+      values[i] =
+          misc::tpMaxWaveNumber * std::exp(-misc::tpLogDz * (misc::numberOfTPGridPoints - i - 1));
+    }
+  }
+  real const& operator[](size_t i) const { return values[i]; };
+
+  private:
+  real values[N]{0};
+};
+
+/**
+ * Inverse Fourier coefficients on the logarithmic grid.
+ */
+template <size_t N>
+class InverseFourierCoefficients {
+  public:
+  constexpr InverseFourierCoefficients() {
+    GridPoints<N> localGridPoints;
+
+    for (size_t i = 1; i < N - 1; ++i) {
+      values[i] = std::sqrt(2 / M_PI) * localGridPoints[i] * misc::tpLogDz;
+    }
+    values[0] = std::sqrt(2 / M_PI) * localGridPoints[0] * (1 + misc::tpLogDz);
```

Comment:
The missing 0.5  should lead to small differences with the master branch, right?

Reply:
Indeed. In particular the error is in the smallest component:
```
(gdb) inspect disc%dynrup%tp_dfinv
$1 = (1.8863643150858097e-07, 6.6425882217799723e-08, 8.9665562162703227e-08, 1.2103584882157821e-07, 1.6338130656424509e-07, 2.2054169565902143e-07, 2.9770015032307667e-07, 4.0185316993030467e-07, 5.4244504078275839e-07, 7.3222421592652987e-07, 9.8839930698886031e-07, 1.3341995099409315e-06, 1.8009809595573516e-06, 2.4310700105351742e-06, 3.2816012655547887e-06, 4.4296983712616928e-06, 5.9794673613526645e-06, 8.0714366823351432e-06, 1.0895299895442135e-05, 1.4707116525044456e-05, 1.9852530775377855e-05, 2.6798113519817424e-05, 3.6173669561147141e-05, 4.8829346459458476e-05, 6.5912723386480073e-05, 8.8972870194560996e-05, 0.00012010081246744448, 0.00016211913950621388, 0.00021883794833910572, 0.00029540033199740393, 0.00039874873990757139, 0.00053825449857406761, 0.00072656757561761077, 0.00098076364124657532, 0.0013238924392870008, 0.0017870678694548382, 0.0024122893033196959, 0.0032562499625074714, 0.0043954776915597401, 0.0059332742754557525, 0.0080090825384880725, 0.010811130605181308, 0.014593499867258468, 0.019699164329178075, 0.026591090471628043, 0.03589421767617744, 0.048452125871238393, 0.065403528853072282, 0.088285529468870846, 0.11917299953506608, 0.16086672304765989, 0.21714736295177328, 0.29311828042235422, 0.39566829248964758, 0.53409632949570884, 0.72095463446379771, 0.97318696319369502, 1.3136649936851528, 1.7732622619301779, 1.1968268412042979)
(gdb) inspect seissol::dr::friction_law::tpInverseFourierCoefficients
$2 = {values = {2.1324118344448284e-07, 6.6425882217799723e-08, 8.9665562162703227e-08, 1.2103584882157821e-07, 1.6338130656424509e-07, 2.2054169565902143e-07, 2.9770015032307667e-07, 4.0185316993030467e-07, 5.4244504078275839e-07, 7.3222421592652987e-07, 9.8839930698886031e-07, 1.3341995099409315e-06, 1.8009809595573516e-06, 2.4310700105351742e-06, 3.2816012655547887e-06, 4.4296983712616928e-06, 5.9794673613526645e-06, 8.0714366823351432e-06, 1.0895299895442135e-05, 1.4707116525044456e-05, 1.9852530775377855e-05, 2.6798113519817424e-05, 3.6173669561147141e-05, 4.8829346459458476e-05, 6.5912723386480073e-05, 8.8972870194560996e-05, 0.00012010081246744448, 0.00016211913950621388, 0.00021883794833910572, 0.00029540033199740393, 0.00039874873990757139, 0.00053825449857406761, 0.00072656757561761077, 0.00098076364124657532, 0.0013238924392870008, 0.0017870678694548382, 0.0024122893033196959, 0.0032562499625074714, 0.0043954776915597401, 0.0059332742754557525, 0.0080090825384880725, 0.010811130605181308, 0.014593499867258468, 0.019699164329178075, 0.026591090471628043, 0.03589421767617744, 0.048452125871238393, 0.065403528853072282, 0.088285529468870846, 0.11917299953506608, 0.16086672304765989, 0.21714736295177328, 0.29311828042235422, 0.39566829248964758, 0.53409632949570884, 0.72095463446379771, 0.97318696319369502, 1.3136649936851528, 1.7732622619301779, 1.1968268412042979}}
```